pkg/resources/server_test.go

```diff
@@ -0,0 +1,327 @@
+	"github.com/Mellanox/k8s-rdma-shared-dev-plugin/pkg/utils"
+	"github.com/fsnotify/fsnotify"
+
+	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1beta1"
+	registerapi "k8s.io/kubernetes/pkg/kubelet/apis/pluginregistration/v1"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/stretchr/testify/mock"
+	"google.golang.org/grpc"
+	"time"
+)
+
+type devPluginListAndWatchServerMock struct {
+	grpc.ServerStream
+}
+
+func (x *devPluginListAndWatchServerMock) Send(m *pluginapi.ListAndWatchResponse) error {
+	return nil
+}
+
+var _ = Describe("Server", func() {
+	Context("newResourcesServer", func() {
+		It("server with plugin watcher enabled", func() {
+			fs := utils.FakeFilesystem{
+				Dirs:     []string{"sys/class/net/ib0/"},
+				Symlinks: map[string]string{"sys/class/net/ib0/device": "../../../0000:02:00.0"},
+			}
+			defer fs.Use()()
+			conf := &types.UserConfig{ResourceName: "test_server", RdmaHcaMax: 100, Devices: []string{"ib0"}}
+			obj, err := newResourceServer(conf, true, "rdma", "socket")
+			Expect(err).ToNot(HaveOccurred())
+			rs := obj.(*resourceServer)
+			Expect(rs.resourceName).To(Equal("rdma/test_server"))
+			Expect(rs.endPoint).To(Equal("test_server.socket"))
+			Expect(rs.watchMode).To(Equal(true))
+			Expect(len(rs.devs)).To(Equal(100))
+		})
+		It("server with plugin watcher disabled", func() {
+			fs := utils.FakeFilesystem{
+				Dirs:     []string{"sys/class/net/ib0/"},
+				Symlinks: map[string]string{"sys/class/net/ib0/device": "../../../0000:02:00.0"},
+			}
+			defer fs.Use()()
+			conf := &types.UserConfig{ResourceName: "test_server", RdmaHcaMax: 100, Devices: []string{"ib0"}}
+			obj, err := newResourceServer(conf, false, "rdma", "socket")
+			Expect(err).ToNot(HaveOccurred())
+			rs := obj.(*resourceServer)
+			Expect(rs.resourceName).To(Equal("rdma/test_server"))
+			Expect(rs.endPoint).To(Equal("test_server.socket"))
+			Expect(rs.watchMode).To(Equal(false))
+			Expect(len(rs.devs)).To(Equal(100))
+		})
+		It("server with plugin with invalid max number of resources", func() {
+			conf := &types.UserConfig{ResourceName: "test_server", RdmaHcaMax: -100, Devices: []string{"ib0"}}
+			obj, err := newResourceServer(conf, true, "rdma", "socket")
+			Expect(err).To(HaveOccurred())
+			Expect(obj).To(BeNil())
+		})
+	})
+	Context("Start", func() {
+		It("start server with plugin watcher enabled", func() {
+			grpcServer := grpc.NewServer([]grpc.ServerOption{}...)
+			rsc := &mocks.ResourceServerConnector{}
+			rsc.On("GetServer", mock.Anything).Return(grpcServer)
+			rsc.On("CreateServer").Return()
+			rsc.On("Listen", mock.Anything, mock.Anything).Return(nil, nil)
+			rsc.On("Serve", mock.Anything).Return()
+			rsc.On("Dial", mock.Anything, mock.Anything).Return(nil, nil)
+			rsc.On("Close", mock.Anything).Return()
+			rs := resourceServer{watchMode: true, rsConnector: rsc}
+			err := rs.Start()
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
can you ensure:
1. dial was called
2. Serve was called
3. RegisterRegistrationServer was called

Reply:
same as above